### DISCUSSION TITLE
How do I define the user name for code-server?

### DISCUSSION THREAD

**Discussion by Pythoneryeah** [coder/code-server#6355] [Q&A]
When I installed code-server using docker, the default user in docker was jovyan. How to customize the user name?
![image](https://github.com/coder/code-server/assets/37642828/3abc47a7-e349-42ab-a79f-52e4d522db86)

The contents of my dockerfile are as follows:
`FROM jupyter/all-spark-notebook:latest
ADD code-server_4.14.1_amd64.deb .
USER root
RUN /bin/sh -c 'dpkg -i code-server_4.14.1_amd64.deb && mkdir -p /opt/code/ext/ && mkdir -p ~/.ipython/kernels/pyspark/'
#ADD config/config1.yaml ~/.config/code-server/
# 设置环境变量
ENV PASSWORD msxf
ENV NODE_OPTIONS --max-old-space-size=8192

# 启动 Code Server 和 Jupyter
CMD ["sh", "-c", "jupyter-notebook --ip='0.0.0.0' --port=8889 --no-browser --allow-root & code-server  --bind-addr 0.0.0.0:9999 --auth password"]
`

**Comment by lg-nineoneninethreenine** [comment]
Have you tried changing the username to your desired username and adding `sudo` to your dpkg command? That is a basic way, but lemme think and I will get a better alternative for you.

**Reply by Pythoneryeah** [reply]
Thank you. I'm ready to give it a try. Do you have a better answer

**Reply by code-asher** [reply]
You might be able to draw inspiration from our own Dockerfile where we change the user: https://github.com/coder/code-server/blob/291682e1c0ef545e89945741f7f05953bd83e9d9/ci/release-image/Dockerfile

**Reply by Pythoneryeah** [reply]
Thank you very much. I will try it according to your suggestions
